.github/workflows/release_mac_freethreading.yml

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# could be deleted when # uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b contains 3.13t, used Quansight-Labs/setup-python@v5 instead
+        
+
+name: MacRelease (freethreading)
+
+on:  # Specifies the event triggering the workflow
+  workflow_call:  # Indicates that this is a reusable workflow
+    inputs:
+      os:
+        required: true
+        type: string
+
+# Use MACOSX_DEPLOYMENT_TARGET=12.0 to produce compatible wheel
+env:
+  MACOSX_DEPLOYMENT_TARGET: "12.0"
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    if: github.event_name != 'pull_request' || startsWith( github.base_ref, 'rel-') || contains( github.event.pull_request.labels.*.name, 'run release CIs')
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        python-version: ['3.13t']
+        target-architecture: ['x86_64', 'arm64']
+
+    steps:
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+    - name: Checkout submodules
+      shell: bash
+      run: |
+        auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+        git submodule sync --recursive
+        git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
```

Comment:
Why is this needed? Can we use actions/checkout for this?

Reply:
That was copy&paste from the non-threading case. Then we could possibly improve it there too.

Reply:
we also use that at [onnx](https://github.com/onnx/onnx/tree/main)/[.github](https://github.com/onnx/onnx/tree/main/.github)/[workflows](https://github.com/onnx/onnx/tree/main/.github/workflows)/release_linux_x86_64.yml

Reply:
I think they should be improved

Reply:
Not at the moment but I think we can start with this one. We can improve it here and copy to the rest later

Reply:
started at https://github.com/onnx/onnx/pull/6762